components/resolved_ts/src/lib.rs

```diff
@@ -103,17 +106,16 @@ impl Resolver {
         // Resolved ts never decrease.
         self.resolved_ts = Some(cmp::max(old_resolved_ts, new_resolved_ts));
 
-        let old_min_ts = self.min_ts.unwrap_or_else(TimeStamp::zero);
         let new_min_ts = if has_lock {
             // If there are some lock, the min_ts must be smaller than
-            // the min start ts, so it must be smaller than
+            // the min start ts, so it guarantees to be smaller than
             // any late arriving commit ts.
             cmp::min(min_start_ts, min_ts)
         } else {
             min_ts
         };
```

Comment:
It seems `new_min_ts` is exactly the same as `new_resolved_ts`. Maybe the code can still be simplified.